modules/database/src/main/java/org/eclipse/xpanse/modules/database/CustomJpaRepository.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.database;
+
+import java.util.Optional;
+import org.springframework.data.repository.NoRepositoryBean;
+import org.springframework.data.repository.Repository;
+
+/**
+ * Custom repository interface for database operations.
+ */
+@NoRepositoryBean
+//CHECKSTYLE OFF: InterfaceTypeParameterName
```

Comment:
@baixinsui instead we can change this rule to allow any uppercase multi-character strings as parameter name. 
https://checkstyle.sourceforge.io/checks/naming/interfacetypeparametername.html
Then the rule will be good for the entire project.